RollupActionSingleNodeTests.testTemporaryIndexDeletedOnRollupFailure

<!--
Please fill out the following information, and ensure you have attempted
to reproduce locally
-->

**Build scan**:
https://gradle-enterprise.elastic.co/s/naqu7x7qvcw6q

**Repro line**:
> ./gradlew ':x-pack:plugin:rollup:test' --tests "org.elasticsearch.xpack.rollup.v2.RollupActionSingleNodeTests.testTemporaryIndexDeletedOnRollupFailure" -Dtests.seed=609D1DD31C78D15E -Dtests.security.manager=true -Dtests.locale=nn-NO -Dtests.timezone=Asia/Bahrain -Druntime.java=11

**Reproduces locally?**:
No

**Applicable branches**:
master

**Failure history**:
Only [once in past 30 days](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testTemporaryIndexDeletedOnRollupFailure),sort:!(process.time-start,desc))). So it could be transient. 

**Failure excerpt**:
```
java.lang.AssertionError: test leaves indices that were not deleted: jzklrm |  
-- | --
  | Expected: [] |  
  | but: was ["jzklrm"]
```

Happened again here: https://gradle-enterprise.elastic.co/s/tjlhzegcbs5um

```
java.lang.AssertionError: test leaves indices that were not deleted: ezolfv |  
-- | --
  | Expected: [] |  
  | but: was ["ezolfv"]
```

I bet there is some assert busy that needs to be added to the test as this is probably done asynchronously some where.

Did not reproduce locally.

This is a different test, but it is the same failure, most likely related:

reproduce:
```
./gradlew ':x-pack:plugin:rollup:test' --tests "org.elasticsearch.xpack.rollup.v2.RollupActionSingleNodeTests.testRollupShardIndexerCleansTempFiles" -Dtests.seed=DFDFD61694502742 -Dtests.security.manager=true -Dtests.locale=en-CA -Dtests.timezone=Brazil/DeNoronha -Druntime.java=11
```

https://gradle-enterprise.elastic.co/s/4x4gsnvprwl5m

```
org.elasticsearch.xpack.rollup.v2.RollupActionSingleNodeTests > testRollupShardIndexerCleansTempFiles FAILED |  
-- | --
  | java.lang.AssertionError: test leaves indices that were not deleted: obgtxw |  
  | Expected: [] |  
  | but: was ["obgtxw"] |  
  | at __randomizedtesting.SeedInfo.seed([DFDFD61694502742:7F5F5CD56BCD71A9]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.elasticsearch.test.ESSingleNodeTestCase.tearDown(ESSingleNodeTestCase.java:134) |  
  | at jdk.internal.reflect.GeneratedMethodAccessor8.invoke(Unknown Source) |  
  | at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) |  
  | at java.base/java.lang.reflect.Method.invoke(Method.java:566) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49) |  
  | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  | at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48) |  
  | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902) |  
  | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41) |  
  | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53) |  
  | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  | at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831) |  
  | at java.base/java.lang.Thread.run(Thread.java:834)
```

--- a/x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java
+++ b/x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java
@@ -63,7 +63,7 @@ import java.util.List;
 /**
  * The master rollup action that coordinates
  *  -  creating rollup temporary index
- *  -  calling {@link TransportRollupIndexerAction} to index rolluped up documents
+ *  -  calling {@link TransportRollupIndexerAction} to index rollup-ed documents
  *  -  cleaning up state
  */
 public class TransportRollupAction extends AcknowledgedTransportMasterNodeAction<RollupAction.Request> {
@@ -139,9 +139,11 @@ public class TransportRollupAction extends AcknowledgedTransportMasterNodeAction
         // 3. run rollup indexer
         // 4. make temp index read-only
         // 5. shrink index
-        // 6. delete temporary index
+        // 6. publish rollup metadata and add rollup index to datastream
+        // 7. delete temporary index
         // at any point if there is an issue, then cleanup temp index
 
+        // 1.
         client.fieldCaps(fieldCapsRequest, ActionListener.wrap(fieldCapsResponse -> {
             RollupActionRequestValidationException validationException = new RollupActionRequestValidationException();
             if (fieldCapsResponse.get().size() == 0) {
@@ -156,6 +158,7 @@ public class TransportRollupAction extends AcknowledgedTransportMasterNodeAction
                 return;
             }
 
+            // 2.
             clusterService.submitStateUpdateTask("rollup create index", new ClusterStateUpdateTask() {
                 @Override
                 public ClusterState execute(ClusterState currentState) throws Exception {
@@ -170,12 +173,16 @@ public class TransportRollupAction extends AcknowledgedTransportMasterNodeAction
 
                 public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
                     // index created
+                    // 3.
                     client.execute(RollupIndexerAction.INSTANCE, rollupIndexerRequest, ActionListener.wrap(indexerResp -> {
                         if (indexerResp.isCreated()) {
+                            // 4.
                             client.admin().indices().updateSettings(updateSettingsReq, ActionListener.wrap(updateSettingsResponse -> {
                                 if (updateSettingsResponse.isAcknowledged()) {
+                                    // 5.
                                     client.admin().indices().resizeIndex(resizeRequest, ActionListener.wrap(resizeResponse -> {
                                         if (resizeResponse.isAcknowledged()) {
+                                            // 6.
                                             publishMetadata(originalIndexName, tmpIndexName, rollupIndexName, listener);
                                         } else {
                                             deleteTmpIndex(originalIndexName, tmpIndexName, listener,

--- a/x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/v2/RollupActionSingleNodeTests.java
+++ b/x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/v2/RollupActionSingleNodeTests.java
@@ -6,7 +6,6 @@
  */
 package org.elasticsearch.xpack.rollup.v2;
 
-import org.apache.lucene.util.LuceneTestCase;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.ResourceAlreadyExistsException;
 import org.elasticsearch.action.ActionListener;
@@ -83,7 +82,6 @@ import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitC
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 
-@LuceneTestCase.AwaitsFix(bugUrl="https://github.com/elastic/elasticsearch/issues/69799")
 public class RollupActionSingleNodeTests extends ESSingleNodeTestCase {
 
     private static final DateFormatter DATE_FORMATTER = DateFormatter.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
@@ -130,7 +128,6 @@ public class RollupActionSingleNodeTests extends ESSingleNodeTestCase {
         super.tearDown();
     }
 
-    @AwaitsFix(bugUrl="https://github.com/elastic/elasticsearch/issues/69506")
     public void testRollupShardIndexerCleansTempFiles() throws IOException {
         // create rollup config and index documents into source index
         RollupActionDateHistogramGroupConfig dateHistogramGroupConfig = randomRollupActionDateHistogramGroupConfig("date_1");
@@ -152,8 +149,8 @@ public class RollupActionSingleNodeTests extends ESSingleNodeTestCase {
         // re-use source index as temp index for test
         RollupShardIndexer indexer = new RollupShardIndexer(client(), indexService, shard.shardId(), config, index, 2);
         indexer.execute();
-        assertThat(indexer.tmpFilesDeleted, equalTo(indexer.tmpFiles));
         // assert that files are deleted
+        assertThat(indexer.tmpFilesDeleted, equalTo(indexer.tmpFiles));
     }
 
     public void testCannotRollupToExistingIndex() throws Exception {
@@ -173,21 +170,14 @@ public class RollupActionSingleNodeTests extends ESSingleNodeTestCase {
         assertThat(exception.getMessage(), containsString("Unable to rollup index [" + index + "]"));
     }
 
-    @AwaitsFix(bugUrl="https://github.com/elastic/elasticsearch/issues/69506")
-    public void testTemporaryIndexDeletedOnRollupFailure() throws Exception {
+    public void testTemporaryIndexCannotBeCreatedAlreadyExists() {
         RollupActionDateHistogramGroupConfig dateHistogramGroupConfig = randomRollupActionDateHistogramGroupConfig("date_1");
-        SourceSupplier sourceSupplier = () -> XContentFactory.jsonBuilder().startObject()
-            .field("date_1", randomDateForInterval(dateHistogramGroupConfig.getInterval()))
-            .field("categorical_1", randomAlphaOfLength(1))
-            .endObject();
         RollupActionConfig config = new RollupActionConfig(
             new RollupActionGroupConfig(dateHistogramGroupConfig, null, new TermsGroupConfig("categorical_1")),
-            Collections.singletonList(new MetricConfig("numeric_non_existent", Collections.singletonList("max"))));
-        bulkIndex(sourceSupplier);
-        expectThrows(ElasticsearchException.class,  () -> rollup(index, rollupIndex, config));
-        // assert that temporary index was removed
-        expectThrows(IndexNotFoundException.class,
-            () -> client().admin().indices().prepareGetIndex().addIndices(".rolluptmp-" + rollupIndex).get());
+            Collections.singletonList(new MetricConfig("numeric_1", Collections.singletonList("max"))));
+        assertTrue(client().admin().indices().prepareCreate(".rolluptmp-" + rollupIndex).get().isAcknowledged());
+        Exception exception = expectThrows(ElasticsearchException.class,  () -> rollup(index, rollupIndex, config));
+        assertThat(exception.getMessage(), containsString("already exists"));
     }
 
     public void testCannotRollupWhileOtherRollupInProgress() throws Exception {

